### PR TITLE
SDI-88 Convert javascript to typescript

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: server/controllers/calendarController.test.ts [hunk 1]
```diff
@@ -18,7 +18,17 @@ jest.mock('jwt-decode', () => {
 })
 
 const calendarData = [
-  { date: new Date(), details: [{ activity: 'Duty Manager', displayType: 'DAY_START', start: new Date() }] },
+  {
+    date: '2022-07-29',
```
Comment (steverendell) [inline server/controllers/calendarController.test.ts:22]: date property is a string!

File: server/helpers/utilities.test.ts [hunk 2]
```diff
@@ -152,44 +151,42 @@ describe('getSnoozeUntil', () => {
     dateNow.mockRestore()
   })
   it('should return a formatted date string if the date is in the future', () => {
-    expect(getSnoozeUntil(new Date('4/2/1994'))).toEqual('3 April 1994')
-    expect(getSnoozeUntil(new Date('9/17/1994'))).toEqual('18 September 1994')
+    expect(utilities.getSnoozeUntil('1994-04-02')).toEqual('3 April 1994')
+    expect(utilities.getSnoozeUntil('1994-09-17')).toEqual('18 September 1994')
```
Comment (steverendell) [inline server/helpers/utilities.test.ts:155]: date should be a string

File: server/helpers/utilities.ts [hunk 3]
```diff
@@ -1,99 +1,66 @@
...
-      error.message.includes('phone_number Not enough digits') ||
-      error.message.includes('phone_number Must not contain letters or symbols') ||
-      error.message.includes('phone_number Too many digits')
-    ) {
-      type =
-        '<p>We do not have a valid mobile phone number for you.</p><p>Please call the Service Desk on 0800 917 5148</p>'
-    }
-  }
-  return type
-}
```
Comment (steverendell) [inline server/helpers/utilities.ts:52]: ^ not used

File: server/middleware/setUpMaintenance.test.ts [hunk 4]
```diff
@@ -23,7 +23,7 @@ describe('maintenance middleware', () => {
     res = { render: renderMock } as unknown as Response
     req = { user: { username: 'user' } } as unknown as Request
 
-    jest.useFakeTimers('modern')
+    jest.useFakeTimers()
```
Comment (steverendell) [inline server/middleware/setUpMaintenance.test.ts:26]: the default in jest v28

File: server/routes/calendarRouter.test.ts [hunk 5]
```diff
@@ -4,20 +4,22 @@ import appWithAllRoutes from './testutils/appSetup'
...
 let app: Express
 
-jest.mock('moment', () => () => jest.requireActual('moment')('2020-03-02T00:00:00.000Z'))
-
 beforeEach(() => {
   app = appWithAllRoutes({ hmppsAuthMFAUser: true })
+  jest.useFakeTimers({
+    doNotFake: ['setImmediate', 'clearImmediate', 'setInterval', 'clearInterval'],
+  })
+  jest.setSystemTime(new Date('2020-03-02'))
```
Comment (steverendell) [inline server/routes/calendarRouter.test.ts:12]: just fake the date otherwise this test hangs

File: server/routes/login.ts [hunk 6]
```diff
@@ -14,9 +14,6 @@ const processError = (error: Error, req: Request, res: Response) => {
     response: res,
     stack: error.stack,
     message: req.user.username,
-    id: req.user.username,
-    authError: true,
-    authErrorText: utilities.getAuthErrorDescription(error),
```
Comment (steverendell) [inline server/routes/login.ts:19]: unused fields

File: server/services/notificationService.ts [hunk 7]
```diff
@@ -24,8 +24,9 @@ export default class NotificationService {
       })
       .then((response) => response.data)
       .catch((error) => {
-        logger.error(getSanitisedError(error), 'notificationService getNotifications')
-        throw error
+        const sanitisedError = getSanitisedError(error)
+        logger.error(sanitisedError, 'notificationService getNotifications')
+        throw sanitisedError
```
Comment (steverendell) [inline server/services/notificationService.ts:29]: fixed potential bug leaking token

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**